zap/src/main/java/org/zaproxy/zap/authentication/PostBasedAuthenticationMethodType.java

```diff
@@ -364,6 +363,10 @@ public WebSession authenticate(
                     LOGGER.debug("Authentication request body: \n" + msg.getRequestBody());
             }
 
+            // Set Content-Length right before sending to avoid any miscalculation from replaced
+            // tokens
+            if (!msg.getRequestHeader().getMethod().equals(HttpRequestHeader.GET))
```

Comment:
- Remove the comment, it's not just about the tokens, it's also needed when there are no tokens.
- Add braces to the if statement.